includes/class-user-manual-sync.php

```diff
@@ -0,0 +1,239 @@
+
+	/**
+	 * The metadata we're syncing.
+	 *
+	 * @var array
+	 */
+	public static $meta_to_sync = [
+		// Social Links.
+		'facebook',
+		'instagram',
+		'linkedin',
+		'myspace',
+		'pinterest',
+		'soundcloud',
+		'tumblr',
+		'twitter',
+		'youtube',
+		'wikipedia',
+
+		// Core bio.
+		'first_name',
+		'last_name',
+		'description',
+
+		// Newspack.
+		'newspack_job_title',
+		'newspack_role',
+		'newspack_employer',
+		'newspack_phone_number',
+
+		// Yoast SEO.
+		'wpseo_title',
+		'wpseo_metadesc',
+		'wpseo_noindex_author',
+		'wpseo_content_analysis_disable',
+		'wpseo_keyword_analysis_disable',
+		'wpseo_inclusive_language_analysis_disable',
+
+		// Simple Local Avatars.
+		'simple_local_avatar',
+		'simple_local_avatar_rating',
+	];
+
+	/**
+	 * Meta keys we watched but we don't want to update in the same way we do with all the others.
+	 *
+	 * @var array
+	 */
+	public static $read_only_meta = [
+		'simple_local_avatar',  // The avatar is sideloaded in a different way.
+	];
+
+	/**
+	 * The user properties we're syncing.
+	 *
+	 * @var array
+	 */
+	public static $user_props = [
+		'display_name',
+		'user_email',
+		'user_url',
+	];
+
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		\add_action( 'edit_user_profile', [ __CLASS__, 'add_manual_sync_button' ] );
+		\add_action( 'show_user_profile', [ __CLASS__, 'add_manual_sync_button' ] );
+		\add_action( 'admin_init', [ __CLASS__, 'process_admin_action' ] );
+		\add_action( 'init', [ __CLASS__, 'register_listener' ] );
+	}
+
+	/**
+	 * Register the listeners to the Newspack Data Events API
+	 *
+	 * @return void
+	 */
+	public static function register_listener() {
+		if ( ! class_exists( 'Newspack\Data_Events' ) ) {
+			return;
+		}
+
+		Data_Events::register_listener( 'newspack_network_manual_sync_user', 'network_manual_sync_user', [ __CLASS__, 'manual_sync_user' ] );
+	}
+
+	/**
+	 * Filters the user data for the event being triggered
+	 *
+	 * @param array $user_data The user data.
+	 * @return array
+	 */
+	public static function manual_sync_user( $user_data ) {
+		$synced_metadata = [];
+		$synced_props    = [];
+
+		// Create an array of all of the synced user meta values.
+		foreach ( self::$meta_to_sync as $key ) {
+			$synced_metadata[ $key ] = $user_data->$key;
+		}
+
+		foreach ( self::$user_props as $key ) {
+			$synced_props[ $key ] = $user_data->$key;
+		}
+
+		return [
+			'email'   => $user_data->user_email,
+			'role'    => array_shift( $user_data->roles ),
+			'user_id' => $user_data->ID,
+			'meta'    => $synced_metadata,
+			'prop'    => $synced_props,
+		];
+	}
+
+	/**
+	 * Create URL to manually sync user.
+	 *
+	 * @param WP_User $user The current WP_User object.
+	 * @return string $url The wp-admin URL.
+	 */
+	public static function get_manual_sync_link( $user ) {
+		$url = add_query_arg(
+			array(
+				'user_id'  => \esc_attr( $user->ID ),
+				'action'   => 'np_network_manual_user_sync',
+				'uid'      => \esc_attr( $user->ID ),
```

Comment:
No need to create another parameter with the same value. You can simply read from `user_id` in you callback down there

Reply:
D'oh! Thanks Leo!